clang/lib/Frontend/CompilerInvocation.cpp

```diff
@@ -2417,6 +2431,9 @@ static void ParseLangArgs(LangOptions &Opts, ArgList &Args, InputKind IK,
   if (Opts.CUDAIsDevice && Args.hasArg(OPT_fcuda_approx_transcendentals))
     Opts.CUDADeviceApproxTranscendentals = 1;
 
+  if (Args.hasArg(OPT_fsycl_is_device))
+      Opts.setSYCLVersion(LangOptions::SYCLVersionList::sycl_1_2_1);
```

Comment:
Will this override any value that the user sets?

Reply:
You are very much correct, doesn't matter 'cos there is only one version now but would be easy to miss that later on.

Reply:
you could move that before https://github.com/Ruyk/llvm/pull/1/files#diff-63bab2358c12cb79fff39bd2de9d3003R2380

That will set the default without overriding